src/parsers/JsonParser.jl

```diff
@@ -120,6 +122,17 @@ function _map_complex(v)
     end
 end
 
+function _map_array(v)
+    if isa(v,Vector) && all(isa.(v,Vector))
+        M = [v...;;]
+        _map_complex.(M)
+    elseif isa(v,Vector)
+        _map_complex.(v)
+    else
+        _map_complex(v)
+    end
```

Comment:
in all of those cases `_map_complex` equals identity, no? So maybe just add
```
elseif v isa Vector && all(isa.(v, Vector)
    [V...;;]
end
```
to `_map_complex`? I think there is no need for another function which then calls `_map_complex` with than does nothing

Reply:
No, not in the case of complex arrays. The problem is that in the json-files complex numbers are saved as dictionaries with keys `"re"` and `"im"` and matrices are saved as arrays of arrays. If we have a complex matrix it is saved as an array of arrays of dictionaries, so first we have to construct a matrix and then apply the complex mapping to every matrix entry. In case of complex vectors we also need broadcasting for the complex mapping. If a parameter is just a complex number it is saved as a dictionary for which we need the case without broadcasting.

Reply:
ah i see, okay then it makes sense